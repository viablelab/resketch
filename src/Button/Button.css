.button {
  background-color: var(--primary-color);
  border-radius: 3px;
  border: 2px solid var(--primary-color);
  box-sizing: border-box;
  color: var(--on-primary-color);
  cursor: pointer;
  font-family: 'Avenir Next', sans-serif;
  font-size: 1rem;
  font-weight: 500;
  height: 48px;
  padding-left: 24px;
  padding-right: 24px;
  text-transform: uppercase;
}

.button:disabled {
  cursor: not-allowed;
  opacity: 0.5;
}

.button--ghost {
  background-color: transparent;
  color: var(--primary-color);
}

.button--text {
  background-color: transparent;
  border-style: none;
  border-width: 0;
  color: var(--primary-color);
}

.button--small {
  font-size: 0.75rem;
  height: 36px;
}
