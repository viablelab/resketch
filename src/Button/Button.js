import styled from 'styled-components';

export default styled.button`
  padding: 12px;
`;
