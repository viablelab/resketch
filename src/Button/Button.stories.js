import React from 'react';
import { storiesOf } from '@storybook/react';
import { boolean, select, text } from '@storybook/addon-knobs';
import Button from './Button';

storiesOf('Button', module).add('Example', () => {
  return React.createElement(Button, {
    children: text('children', 'Click Me'),
    disabled: boolean('disabled', false),
    look: select('look', [null, 'ghost', 'text']) || undefined,
  });
});
