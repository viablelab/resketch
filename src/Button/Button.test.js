import React from 'react';
import renderer from 'react-test-renderer';
import Button from './Button';

test('renders properly', () => {
  const tree = renderer.create(<Button>Button label</Button>).toJSON();

  expect(tree).toMatchSnapshot();
});
