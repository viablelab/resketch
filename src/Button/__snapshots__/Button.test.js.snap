// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders properly 1`] = `
.c0 {
  padding: 12px;
}

<button
  className="c0"
>
  Button label
</button>
`;
